SYCL/ESIMD/api/functional/common.hpp

```diff
@@ -55,6 +55,24 @@ using shared_allocator = sycl::usm_allocator<DataT, sycl::usm::alloc::shared>;
 template <typename DataT>
 using shared_vector = std::vector<DataT, shared_allocator<DataT>>;
 
+template <typename T>
+constexpr bool should_run_test_with_current_datatype(sycl::device &&device) {
```

Comment:
```suggestion
constexpr bool should_run_test_with_current_datatype(const sycl::device &device) {
```
We are not modifying it or passing along anyway, so why limit usability to the r-values only? What do you think?

Reply:
this comment was applied [here](https://github.com/intel/llvm-test-suite/pull/839/commits/1c069b5f4a52017db43d26010feca6a451457ef9#diff-ef0766a5c874c909fd25e4eec71893c4614c91112485144e4efcdab4d856d0a0L59-R72)